test/spectral/tests/custom-rules/error-response.test.js

```diff
@@ -0,0 +1,492 @@
+              '400': {
+                $ref: '#/components/responses/ErrorResponse'
+              },
+              '404': {
+                $ref: '#/components/responses/AcceptableErrorResponse'
+              },
+              '500': {
+                $ref: '#/components/responses/ErrorResponse'
+              }
+            }
+          }
+        }
+      },
+      components: {
+        responses: {
+          AcceptableErrorResponse: {
+            content: {
+              'application/json': {
+                schema: {
+                  $ref: '#/components/schemas/AcceptableErrorModel'
+                }
+              }
+            }
+          },
+          ErrorResponse: {
+            content: {
+              'application/json': {
+                schema: {
+                  $ref: '#/components/schemas/ErrorContainerModel'
+                }
+              }
+            }
+          }
+        },
+        schemas: {
+          AcceptableErrorModel: {
+            type: 'object',
+            properties: {
+              error: {
+                $ref: '#/components/schemas/ErrorModel'
+              },
+              trace: {
+                type: 'string',
+                format: 'uuid'
+              }
+            }
+          },
+          ErrorContainerModel: {
+            type: 'object',
+            properties: {
+              errors: {
+                type: 'array',
+                items: {
+                  $ref: '#/components/schemas/ErrorModel'
+                }
+              },
+              trace: {
+                type: 'string',
+                format: 'uuid'
+              },
+              status_code: {
+                type: 'integer'
+              }
+            }
+          },
+          ErrorModel: {
+            type: 'object',
+            properties: {
+              code: {
+                type: 'string',
+                enum: ['error_1', 'error_2']
+              },
+              message: {
+                type: 'string'
+              },
+              more_info: {
+                type: 'string',
+                description: 'url with link to more information about the error'
+              },
+              target: {
+                $ref: '#/components/schemas/ErrorTargetModel'
+              }
+            }
+          },
+          ErrorTargetModel: {
+            type: {
+              type: 'string',
+              enum: ['field', 'parameter', 'header']
+            },
+            name: {
+              type: 'string',
+              description: 'name of the problematic field that caused the error'
+            }
+          }
+        }
+      }
+    };
+
+    res = await inCodeValidator(spec, true);
+  });
+
+  it('should not error when content object provided', function() {
+    const expectedWarnings = res.warnings.filter(
+      warn => warn.message === 'Error response should have a content field'
+    );
+    expect(expectedWarnings.length).toBe(0);
+  });
```

Comment:
This style is funky but I don't dislike it!

Reply:
Yes, the goal with this is to isolate tests a bit more to prevent new rules causing cascading test failures.

Reply:
The downside for the negative test is that it will cause false positives if the error message is changed, for example. That may be enough reason to take a different approach with the negative tests but still use this technique for the positive tests.